### ISSUE TITLE
Thanos query shows no result

### ISSUE THREAD

**Issue by argonmist** [coreos/prometheus-operator#1582]
**What did you do?**
I modified some yaml files last week to enable thanos sidecar, everything worked fine, and I re-clone repo today with the same setting, thanos UI can not select the `metric at cursor` bar, but I tried thanos example, it's ok. 
Here are all the yaml files I modified:

prometheus-prometheus.yaml
```
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  labels:
    prometheus: k8s
  name: k8s
  namespace: monitoring
spec:
  podMetadata:
    labels:
      thanos-peer: 'true'
  alerting:
    alertmanagers:
    - name: alertmanager-main
      namespace: monitoring
      port: web
  baseImage: quay.io/prometheus/prometheus
  nodeSelector:
    beta.kubernetes.io/os: linux
  replicas: 2
  resources:
    requests:
      memory: 400Mi
  ruleSelector:
    matchLabels:
      prometheus: k8s
      role: alert-rules
  serviceAccountName: prometheus-k8s
  serviceMonitorSelector:
    matchExpressions:
    - key: k8s-app
      operator: Exists
  version: v2.3.1
  thanos:
    peers: thanos-peers.default.svc:10900
```

prometheus-service.yaml
```
apiVersion: v1
kind: Service
metadata:
  labels:
    k8s-app: prometheus
    prometheus: k8s
  name: prometheus-k8s
  namespace: monitoring
spec:
  type: NodePort
  ports:
  - name: web
    port: 9090
    protocol: TCP
    targetPort: web
  selector:
    app: prometheus
    prometheus: k8s
```

prometheus-serviceMonitor.yaml
```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    k8s-app: prometheus
  name: prometheus
  namespace: monitoring
spec:
  endpoints:
  - interval: 30s
    port: web
  selector:
    matchLabels:
      k8s-app: prometheus
      prometheus: k8s
```

querier-deployment.yaml
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: thanos-query
  namespace: monitoring
  labels:
    app: thanos-query
    thanos-peer: "true"
spec:
  replicas: 1
  selector:
    matchLabels:
      app: thanos-query
      thanos-peer: "true"
  template:
    metadata:
      labels:
        app: thanos-query
        thanos-peer: "true"
    spec:
      containers:
      - name: thanos-query
        image: improbable/thanos:v0.1.0-rc.1
        args:
        - "query"
        - "--log.level=debug"
        - "--query.replica-label=prometheus_replica"
        - "--cluster.peers=thanos-peers.default.svc:10900"
        ports:
        - name: http
          containerPort: 10902
        - name: grpc
          containerPort: 10901
        - name: cluster
          containerPort: 10900
```

querier-service.yaml
```
apiVersion: v1
kind: Service
metadata:
  labels:
    app: thanos-query
  name: thanos-query
  namespace: monitoring
spec:
  type: NodePort
  selector:
    app: thanos-query
  ports:
  - port: 9090
    protocol: TCP
    targetPort: http
    name: http-query
```

thanos-peers-svc.yaml
```
apiVersion: v1
kind: Service
metadata:
  name: thanos-peers
  namespace: monitoring
spec:
  type: ClusterIP
  clusterIP: None
  ports:
  - name: cluster
    port: 10900
    targetPort: cluster
  selector:
    thanos-peer: "true"
```
**What did you expect to see?**
Thanos query works correctly

**Environment**
 kubectl version 
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:21:50Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:13:31Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

kubeadm

**Comment by argonmist** [comment]
Thanks for the information, I've tried just restart Prometheus servers, but still got the bug, I'll try again if the new version released.

**Comment by jpke** [comment]
any update on this? trying to setup thanos to push prometheus data to s3 using https://github.com/coreos/prometheus-operator/tree/master/example/thanos

**Comment by brancz** [comment]
Are both of you using the latest release candidate? If yes, could you share the metrics output of one of the queriers? (just port-forward the metrics port and curl the `/metrics` endpoint) The metrics tell us more about the peers that have been discovered (or not if they were not discovered properly).

**Comment by jpke** [comment]
I believe I'm using the latest release candidate. Here's the deployment yaml:
```apiVersion: apps/v1
kind: Deployment
metadata:
  name: thanos-query
  labels:
    app: thanos-query
    thanos-peer: "true"
spec:
  replicas: 1
  selector:
    matchLabels:
      app: thanos-query
      thanos-peer: "true"
  template:
    metadata:
      labels:
        app: thanos-query
        thanos-peer: "true"
    spec:
      containers:
      - name: thanos-query
        image: improbable/thanos:v0.1.0-rc.2
        args:
        - "query"
        - "--log.level=debug"
        - "--query.replica-label=prometheus_replica"
        - "--cluster.peers=thanos-peers.default.svc:10900"
        ports:
        - name: http
          containerPort: 10902
        - name: grpc
          containerPort: 10901
        - name: cluster
          containerPort: 10900```

Here's the result of the `/metrics` endpoint:
```▶ curl localhost:10902/metrics
# HELP go_gc_duration_seconds A summary of the GC invocation durations.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 2.3049e-05
go_gc_duration_seconds{quantile="0.25"} 3.3912e-05
go_gc_duration_seconds{quantile="0.5"} 5.202e-05
go_gc_duration_seconds{quantile="0.75"} 0.000114258
go_gc_duration_seconds{quantile="1"} 0.001657333
go_gc_duration_seconds_sum 0.002692418
go_gc_duration_seconds_count 16
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 49
# HELP go_memstats_alloc_bytes Number of bytes allocated and still in use.
# TYPE go_memstats_alloc_bytes gauge
go_memstats_alloc_bytes 5.61524e+06
# HELP go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
# TYPE go_memstats_alloc_bytes_total counter
go_memstats_alloc_bytes_total 6.1953272e+07
# HELP go_memstats_buck_hash_sys_bytes Number of bytes used by the profiling bucket hash table.
# TYPE go_memstats_buck_hash_sys_bytes gauge
go_memstats_buck_hash_sys_bytes 1.474401e+06
# HELP go_memstats_frees_total Total number of frees.
# TYPE go_memstats_frees_total counter
go_memstats_frees_total 68070
# HELP go_memstats_gc_sys_bytes Number of bytes used for garbage collection system metadata.
# TYPE go_memstats_gc_sys_bytes gauge
go_memstats_gc_sys_bytes 675840
# HELP go_memstats_heap_alloc_bytes Number of heap bytes allocated and still in use.
# TYPE go_memstats_heap_alloc_bytes gauge
go_memstats_heap_alloc_bytes 5.61524e+06
# HELP go_memstats_heap_idle_bytes Number of heap bytes waiting to be used.
# TYPE go_memstats_heap_idle_bytes gauge
go_memstats_heap_idle_bytes 7.200768e+06
# HELP go_memstats_heap_inuse_bytes Number of heap bytes that are in use.
# TYPE go_memstats_heap_inuse_bytes gauge
go_memstats_heap_inuse_bytes 6.660096e+06
# HELP go_memstats_heap_objects Number of allocated objects.
# TYPE go_memstats_heap_objects gauge
go_memstats_heap_objects 24559
# HELP go_memstats_heap_released_bytes_total Total number of heap bytes released to OS.
# TYPE go_memstats_heap_released_bytes_total counter
go_memstats_heap_released_bytes_total 1.744896e+06
# HELP go_memstats_heap_sys_bytes Number of heap bytes obtained from system.
# TYPE go_memstats_heap_sys_bytes gauge
go_memstats_heap_sys_bytes 1.3860864e+07
# HELP go_memstats_last_gc_time_seconds Number of seconds since 1970 of last garbage collection.
# TYPE go_memstats_last_gc_time_seconds gauge
go_memstats_last_gc_time_seconds 1.5330510405681014e+09
# HELP go_memstats_lookups_total Total number of pointer lookups.
# TYPE go_memstats_lookups_total counter
go_memstats_lookups_total 294
# HELP go_memstats_mallocs_total Total number of mallocs.
# TYPE go_memstats_mallocs_total counter
go_memstats_mallocs_total 92629
# HELP go_memstats_mcache_inuse_bytes Number of bytes in use by mcache structures.
# TYPE go_memstats_mcache_inuse_bytes gauge
go_memstats_mcache_inuse_bytes 3472
# HELP go_memstats_mcache_sys_bytes Number of bytes used for mcache structures obtained from system.
# TYPE go_memstats_mcache_sys_bytes gauge
go_memstats_mcache_sys_bytes 16384
# HELP go_memstats_mspan_inuse_bytes Number of bytes in use by mspan structures.
# TYPE go_memstats_mspan_inuse_bytes gauge
go_memstats_mspan_inuse_bytes 71592
# HELP go_memstats_mspan_sys_bytes Number of bytes used for mspan structures obtained from system.
# TYPE go_memstats_mspan_sys_bytes gauge
go_memstats_mspan_sys_bytes 114688
# HELP go_memstats_next_gc_bytes Number of heap bytes when next garbage collection will take place.
# TYPE go_memstats_next_gc_bytes gauge
go_memstats_next_gc_bytes 7.851568e+06
# HELP go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE go_memstats_other_sys_bytes gauge
go_memstats_other_sys_bytes 682391
# HELP go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE go_memstats_stack_inuse_bytes gauge
go_memstats_stack_inuse_bytes 819200
# HELP go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE go_memstats_stack_sys_bytes gauge
go_memstats_stack_sys_bytes 819200
# HELP go_memstats_sys_bytes Number of bytes obtained by system. Sum of all system allocations.
# TYPE go_memstats_sys_bytes gauge
go_memstats_sys_bytes 1.7643768e+07
# HELP http_request_duration_microseconds The HTTP request latencies in microseconds.
# TYPE http_request_duration_microseconds summary
http_request_duration_microseconds{handler="flags",quantile="0.5"} 495.908
http_request_duration_microseconds{handler="flags",quantile="0.9"} 495.908
http_request_duration_microseconds{handler="flags",quantile="0.99"} 495.908
http_request_duration_microseconds_sum{handler="flags"} 495.908
http_request_duration_microseconds_count{handler="flags"} 1
http_request_duration_microseconds{handler="graph",quantile="0.5"} 921.407
http_request_duration_microseconds{handler="graph",quantile="0.9"} 921.407
http_request_duration_microseconds{handler="graph",quantile="0.99"} 921.407
http_request_duration_microseconds_sum{handler="graph"} 921.407
http_request_duration_microseconds_count{handler="graph"} 1
http_request_duration_microseconds{handler="label_values",quantile="0.5"} 59.571
http_request_duration_microseconds{handler="label_values",quantile="0.9"} 59.571
http_request_duration_microseconds{handler="label_values",quantile="0.99"} 59.571
http_request_duration_microseconds_sum{handler="label_values"} 59.571
http_request_duration_microseconds_count{handler="label_values"} 1
http_request_duration_microseconds{handler="options",quantile="0.5"} NaN
http_request_duration_microseconds{handler="options",quantile="0.9"} NaN
http_request_duration_microseconds{handler="options",quantile="0.99"} NaN
http_request_duration_microseconds_sum{handler="options"} 0
http_request_duration_microseconds_count{handler="options"} 0
http_request_duration_microseconds{handler="query",quantile="0.5"} 267.169
http_request_duration_microseconds{handler="query",quantile="0.9"} 376.282
http_request_duration_microseconds{handler="query",quantile="0.99"} 376.282
http_request_duration_microseconds_sum{handler="query"} 643.451
http_request_duration_microseconds_count{handler="query"} 2
http_request_duration_microseconds{handler="query_range",quantile="0.5"} NaN
http_request_duration_microseconds{handler="query_range",quantile="0.9"} NaN
http_request_duration_microseconds{handler="query_range",quantile="0.99"} NaN
http_request_duration_microseconds_sum{handler="query_range"} 0
http_request_duration_microseconds_count{handler="query_range"} 0
http_request_duration_microseconds{handler="series",quantile="0.5"} NaN
http_request_duration_microseconds{handler="series",quantile="0.9"} NaN
http_request_duration_microseconds{handler="series",quantile="0.99"} NaN
http_request_duration_microseconds_sum{handler="series"} 0
http_request_duration_microseconds_count{handler="series"} 0
http_request_duration_microseconds{handler="static",quantile="0.5"} 322.194
http_request_duration_microseconds{handler="static",quantile="0.9"} 3905.828
http_request_duration_microseconds{handler="static",quantile="0.99"} 4670.632
http_request_duration_microseconds_sum{handler="static"} 28271.778999999995
http_request_duration_microseconds_count{handler="static"} 22
http_request_duration_microseconds{handler="status",quantile="0.5"} 803.092
http_request_duration_microseconds{handler="status",quantile="0.9"} 803.092
http_request_duration_microseconds{handler="status",quantile="0.99"} 803.092
http_request_duration_microseconds_sum{handler="status"} 803.092
http_request_duration_microseconds_count{handler="status"} 1
# HELP http_request_size_bytes The HTTP request sizes in bytes.
# TYPE http_request_size_bytes summary
http_request_size_bytes{handler="flags",quantile="0.5"} 433
http_request_size_bytes{handler="flags",quantile="0.9"} 433
http_request_size_bytes{handler="flags",quantile="0.99"} 433
http_request_size_bytes_sum{handler="flags"} 433
http_request_size_bytes_count{handler="flags"} 1
http_request_size_bytes{handler="graph",quantile="0.5"} 360
http_request_size_bytes{handler="graph",quantile="0.9"} 360
http_request_size_bytes{handler="graph",quantile="0.99"} 360
http_request_size_bytes_sum{handler="graph"} 360
http_request_size_bytes_count{handler="graph"} 1
http_request_size_bytes{handler="label_values",quantile="0.5"} 399
http_request_size_bytes{handler="label_values",quantile="0.9"} 399
http_request_size_bytes{handler="label_values",quantile="0.99"} 399
http_request_size_bytes_sum{handler="label_values"} 399
http_request_size_bytes_count{handler="label_values"} 1
http_request_size_bytes{handler="options",quantile="0.5"} NaN
http_request_size_bytes{handler="options",quantile="0.9"} NaN
http_request_size_bytes{handler="options",quantile="0.99"} NaN
http_request_size_bytes_sum{handler="options"} 0
http_request_size_bytes_count{handler="options"} 0
http_request_size_bytes{handler="query",quantile="0.5"} 396
http_request_size_bytes{handler="query",quantile="0.9"} 461
http_request_size_bytes{handler="query",quantile="0.99"} 461
http_request_size_bytes_sum{handler="query"} 857
http_request_size_bytes_count{handler="query"} 2
http_request_size_bytes{handler="query_range",quantile="0.5"} NaN
http_request_size_bytes{handler="query_range",quantile="0.9"} NaN
http_request_size_bytes{handler="query_range",quantile="0.99"} NaN
http_request_size_bytes_sum{handler="query_range"} 0
http_request_size_bytes_count{handler="query_range"} 0
http_request_size_bytes{handler="series",quantile="0.5"} NaN
http_request_size_bytes{handler="series",quantile="0.9"} NaN
http_request_size_bytes{handler="series",quantile="0.99"} NaN
http_request_size_bytes_sum{handler="series"} 0
http_request_size_bytes_count{handler="series"} 0
http_request_size_bytes{handler="static",quantile="0.5"} 365
http_request_size_bytes{handler="static",quantile="0.9"} 405
http_request_size_bytes{handler="static",quantile="0.99"} 468
http_request_size_bytes_sum{handler="static"} 8308
http_request_size_bytes_count{handler="static"} 22
http_request_size_bytes{handler="status",quantile="0.5"} 396
http_request_size_bytes{handler="status",quantile="0.9"} 396
http_request_size_bytes{handler="status",quantile="0.99"} 396
http_request_size_bytes_sum{handler="status"} 396
http_request_size_bytes_count{handler="status"} 1
# HELP http_requests_total Total number of HTTP requests made.
# TYPE http_requests_total counter
http_requests_total{code="200",handler="flags",method="get"} 1
http_requests_total{code="200",handler="graph",method="get"} 1
http_requests_total{code="200",handler="label_values",method="get"} 1
http_requests_total{code="200",handler="query",method="get"} 2
http_requests_total{code="200",handler="static",method="get"} 22
http_requests_total{code="200",handler="status",method="get"} 1
# HELP http_response_size_bytes The HTTP response sizes in bytes.
# TYPE http_response_size_bytes summary
http_response_size_bytes{handler="flags",quantile="0.5"} 2584
http_response_size_bytes{handler="flags",quantile="0.9"} 2584
http_response_size_bytes{handler="flags",quantile="0.99"} 2584
http_response_size_bytes_sum{handler="flags"} 2584
http_response_size_bytes_count{handler="flags"} 1
http_response_size_bytes{handler="graph",quantile="0.5"} 4455
http_response_size_bytes{handler="graph",quantile="0.9"} 4455
http_response_size_bytes{handler="graph",quantile="0.99"} 4455
http_response_size_bytes_sum{handler="graph"} 4455
http_response_size_bytes_count{handler="graph"} 1
http_response_size_bytes{handler="label_values",quantile="0.5"} 33
http_response_size_bytes{handler="label_values",quantile="0.9"} 33
http_response_size_bytes{handler="label_values",quantile="0.99"} 33
http_response_size_bytes_sum{handler="label_values"} 33
http_response_size_bytes_count{handler="label_values"} 1
http_response_size_bytes{handler="options",quantile="0.5"} NaN
http_response_size_bytes{handler="options",quantile="0.9"} NaN
http_response_size_bytes{handler="options",quantile="0.99"} NaN
http_response_size_bytes_sum{handler="options"} 0
http_response_size_bytes_count{handler="options"} 0
http_response_size_bytes{handler="query",quantile="0.5"} 64
http_response_size_bytes{handler="query",quantile="0.9"} 95
http_response_size_bytes{handler="query",quantile="0.99"} 95
http_response_size_bytes_sum{handler="query"} 159
http_response_size_bytes_count{handler="query"} 2
http_response_size_bytes{handler="query_range",quantile="0.5"} NaN
http_response_size_bytes{handler="query_range",quantile="0.9"} NaN
http_response_size_bytes{handler="query_range",quantile="0.99"} NaN
http_response_size_bytes_sum{handler="query_range"} 0
http_response_size_bytes_count{handler="query_range"} 0
http_response_size_bytes{handler="series",quantile="0.5"} NaN
http_response_size_bytes{handler="series",quantile="0.9"} NaN
http_response_size_bytes{handler="series",quantile="0.99"} NaN
http_response_size_bytes_sum{handler="series"} 0
http_response_size_bytes_count{handler="series"} 0
http_response_size_bytes{handler="static",quantile="0.5"} 12881
http_response_size_bytes{handler="static",quantile="0.9"} 113498
http_response_size_bytes{handler="static",quantile="0.99"} 184190
http_response_size_bytes_sum{handler="static"} 889540
http_response_size_bytes_count{handler="static"} 22
http_response_size_bytes{handler="status",quantile="0.5"} 3574
http_response_size_bytes{handler="status",quantile="0.9"} 3574
http_response_size_bytes{handler="status",quantile="0.99"} 3574
http_response_size_bytes_sum{handler="status"} 3574
http_response_size_bytes_count{handler="status"} 1
# HELP prometheus_engine_queries The current number of queries being executed or waiting.
# TYPE prometheus_engine_queries gauge
prometheus_engine_queries 0
# HELP prometheus_engine_queries_concurrent_max The max number of concurrent queries.
# TYPE prometheus_engine_queries_concurrent_max gauge
prometheus_engine_queries_concurrent_max 20
# HELP prometheus_engine_query_duration_seconds Query timings
# TYPE prometheus_engine_query_duration_seconds summary
prometheus_engine_query_duration_seconds{slice="inner_eval",quantile="0.5"} 2.249e-06
prometheus_engine_query_duration_seconds{slice="inner_eval",quantile="0.9"} 9.313e-06
prometheus_engine_query_duration_seconds{slice="inner_eval",quantile="0.99"} 9.313e-06
prometheus_engine_query_duration_seconds_sum{slice="inner_eval"} 1.1562e-05
prometheus_engine_query_duration_seconds_count{slice="inner_eval"} 2
prometheus_engine_query_duration_seconds{slice="prepare_time",quantile="0.5"} 1.4304e-05
prometheus_engine_query_duration_seconds{slice="prepare_time",quantile="0.9"} 0.000150145
prometheus_engine_query_duration_seconds{slice="prepare_time",quantile="0.99"} 0.000150145
prometheus_engine_query_duration_seconds_sum{slice="prepare_time"} 0.000164449
prometheus_engine_query_duration_seconds_count{slice="prepare_time"} 2
prometheus_engine_query_duration_seconds{slice="result_append",quantile="0.5"} NaN
prometheus_engine_query_duration_seconds{slice="result_append",quantile="0.9"} NaN
prometheus_engine_query_duration_seconds{slice="result_append",quantile="0.99"} NaN
prometheus_engine_query_duration_seconds_sum{slice="result_append"} 0
prometheus_engine_query_duration_seconds_count{slice="result_append"} 0
prometheus_engine_query_duration_seconds{slice="result_sort",quantile="0.5"} NaN
prometheus_engine_query_duration_seconds{slice="result_sort",quantile="0.9"} NaN
prometheus_engine_query_duration_seconds{slice="result_sort",quantile="0.99"} NaN
prometheus_engine_query_duration_seconds_sum{slice="result_sort"} 0
prometheus_engine_query_duration_seconds_count{slice="result_sort"} 0
# HELP query_memberlist_gossip query_memberlist_gossip
# TYPE query_memberlist_gossip summary
query_memberlist_gossip{quantile="0.5"} 0.00698699988424778
query_memberlist_gossip{quantile="0.9"} 0.008616000413894653
query_memberlist_gossip{quantile="0.99"} 0.008616000413894653
query_memberlist_gossip_sum 0.5001780022867024
query_memberlist_gossip_count 61
# HELP query_thanos_query_576f49b597_zr227_runtime_alloc_bytes query_thanos_query_576f49b597_zr227_runtime_alloc_bytes
# TYPE query_thanos_query_576f49b597_zr227_runtime_alloc_bytes gauge
query_thanos_query_576f49b597_zr227_runtime_alloc_bytes 5.547968e+06
# HELP query_thanos_query_576f49b597_zr227_runtime_free_count query_thanos_query_576f49b597_zr227_runtime_free_count
# TYPE query_thanos_query_576f49b597_zr227_runtime_free_count gauge
query_thanos_query_576f49b597_zr227_runtime_free_count 67997
# HELP query_thanos_query_576f49b597_zr227_runtime_heap_objects query_thanos_query_576f49b597_zr227_runtime_heap_objects
# TYPE query_thanos_query_576f49b597_zr227_runtime_heap_objects gauge
query_thanos_query_576f49b597_zr227_runtime_heap_objects 24192
# HELP query_thanos_query_576f49b597_zr227_runtime_malloc_count query_thanos_query_576f49b597_zr227_runtime_malloc_count
# TYPE query_thanos_query_576f49b597_zr227_runtime_malloc_count gauge
query_thanos_query_576f49b597_zr227_runtime_malloc_count 92189
# HELP query_thanos_query_576f49b597_zr227_runtime_num_goroutines query_thanos_query_576f49b597_zr227_runtime_num_goroutines
# TYPE query_thanos_query_576f49b597_zr227_runtime_num_goroutines gauge
query_thanos_query_576f49b597_zr227_runtime_num_goroutines 26
# HELP query_thanos_query_576f49b597_zr227_runtime_sys_bytes query_thanos_query_576f49b597_zr227_runtime_sys_bytes
# TYPE query_thanos_query_576f49b597_zr227_runtime_sys_bytes gauge
query_thanos_query_576f49b597_zr227_runtime_sys_bytes 1.7643768e+07
# HELP query_thanos_query_576f49b597_zr227_runtime_total_gc_pause_ns query_thanos_query_576f49b597_zr227_runtime_total_gc_pause_ns
# TYPE query_thanos_query_576f49b597_zr227_runtime_total_gc_pause_ns gauge
query_thanos_query_576f49b597_zr227_runtime_total_gc_pause_ns 2.692418e+06
# HELP query_thanos_query_576f49b597_zr227_runtime_total_gc_runs query_thanos_query_576f49b597_zr227_runtime_total_gc_runs
# TYPE query_thanos_query_576f49b597_zr227_runtime_total_gc_runs gauge
query_thanos_query_576f49b597_zr227_runtime_total_gc_runs 16
# HELP thanos_build_info A metric with a constant '1' value labeled by version, revision, branch, and goversion from which thanos was built.
# TYPE thanos_build_info gauge
thanos_build_info{branch="HEAD",goversion="go1.10.3",revision="53e4d69ebba12baef46543468992619fcfd524ef",version="0.1.0-rc.2"} 1
# HELP thanos_cluster_members Number indicating current number of members in cluster.
# TYPE thanos_cluster_members gauge
thanos_cluster_members 1
# HELP thanos_gossip_messages_received_total Total gossip NotifyMsg calls.
# TYPE thanos_gossip_messages_received_total counter
thanos_gossip_messages_received_total 0
# HELP thanos_grpc_req_panics_recovered_total Total number of gRPC requests recovered from internal panic.
# TYPE thanos_grpc_req_panics_recovered_total counter
thanos_grpc_req_panics_recovered_total 0
# HELP thanos_query_api_instant_query_duration_seconds Time it takes to perform instant query on promEngine backed up with thanos querier.
# TYPE thanos_query_api_instant_query_duration_seconds histogram
thanos_query_api_instant_query_duration_seconds_bucket{le="0.05"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="0.1"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="0.25"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="0.6"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="1"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="2"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="3.5"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="5"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="7.5"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="10"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="15"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="20"} 2
thanos_query_api_instant_query_duration_seconds_bucket{le="+Inf"} 2
thanos_query_api_instant_query_duration_seconds_sum 0.000326199
thanos_query_api_instant_query_duration_seconds_count 2
# HELP thanos_query_api_range_query_duration_seconds Time it takes to perform range query on promEngine backed up with thanos querier.
# TYPE thanos_query_api_range_query_duration_seconds histogram
thanos_query_api_range_query_duration_seconds_bucket{le="0.05"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="0.1"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="0.25"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="0.6"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="1"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="2"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="3.5"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="5"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="7.5"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="10"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="15"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="20"} 0
thanos_query_api_range_query_duration_seconds_bucket{le="+Inf"} 0
thanos_query_api_range_query_duration_seconds_sum 0
thanos_query_api_range_query_duration_seconds_count 0
# HELP thanos_store_nodes_grpc_connections Number indicating current number of gRPC connection to store nodes. This indicates also to how many stores query node have access to.
# TYPE thanos_store_nodes_grpc_connections gauge
thanos_store_nodes_grpc_connections 0```

**Comment by brancz** [comment]
@jpke those metrics tell us that the querier does not have any prometheus nodes to query data from. Did you try to deploy the exact manifests from this repository (unmodified)? If you did modify them, could you first try the exact ones? Thanks!

**Comment by ervikrant06** [comment]
I am facing the same issue. Before injecting the thanos sidecar everything was working fine for me in prometheus UI. After injecting the sidecar, prometheus stopped showing the query results in UI. Implemented thanos-query but thanos UI is also not showing any result. 

Manifest which is given in repo doesn't make clear how can we use an alternate image because in manifest no where we have specified the image. 

~~~
kubectl get pod -n monitoring -l app=prometheus
NAME               READY     STATUS    RESTARTS   AGE
prometheus-k8s-0   4/4       Running   1          3h
prometheus-k8s-1   4/4       Running   2          3h
~~~

I checked the logs of thanos sidecar it was throwing some DNS resolution error messages with port. But after stop/start of thanos sidecar that error has gone away. Following are the logs from sidecar.

~~~
kubectl logs prometheus-k8s-0 -c thanos-sidecar  -n monitoring
level=info ts=2019-01-07T14:40:58.243987719Z caller=flags.go:68 msg="StoreAPI address that will be propagated through gossip" address=172.17.0.23:10901
level=info ts=2019-01-07T14:40:58.258679933Z caller=sidecar.go:233 msg="No GCS or S3 bucket was configured, uploads will be disabled"
level=info ts=2019-01-07T14:40:58.258727853Z caller=sidecar.go:270 msg="starting sidecar" peer=
level=info ts=2019-01-07T14:40:58.26026463Z caller=main.go:244 msg="Listening for metrics" address=0.0.0.0:10902
level=info ts=2019-01-07T14:40:58.260624868Z caller=sidecar.go:215 component=sidecar msg="Listening for StoreAPI gRPC" address=[172.17.0.23]:10901
level=warn ts=2019-01-07T14:41:08.288970764Z caller=cluster.go:322 component=cluster NumMembers=1 msg="I appear to be alone in the cluster"
level=warn ts=2019-01-07T14:41:18.288756092Z caller=cluster.go:322 component=cluster NumMembers=1 msg="I appear to be alone in the cluster"
level=warn ts=2019-01-07T14:41:28.288318825Z caller=cluster.go:322 component=cluster NumMembers=1 msg="I appear to be alone in the cluster"
level=warn ts=2019-01-07T14:41:38.288392248Z caller=cluster.go:322 component=cluster NumMembers=1 msg="I appear to be alone in the cluster"
level=info ts=2019-01-07T14:43:58.266078327Z caller=reloader.go:189 component=reloader msg="Prometheus reload triggered" cfg_in= cfg_out= rule_dirs=
~~~
